client_plugin/drivers/shared/etcdops.go

```diff
@@ -0,0 +1,269 @@
+		return nil
+	}
+
+	// check my local role
+	node, _, err := cli.NodeInspectWithRaw(ctx, nodeID)
+	if err != nil {
+		log.WithFields(log.Fields{"nodeID": nodeID,
+			"error": err}).Error("Failed to inspect node ")
+		return err
+	}
+
+	// get the IP address of current node
+	addr := info.Swarm.NodeAddr
+
+	// if leader, proceed to start ETCD cluster
+	if node.ManagerStatus.Leader {
+		log.WithFields(
+			log.Fields{"nodeID": nodeID},
+		).Info("Swarm node role: leader, start etcd cluster")
+		err = startEtcdCluster(addr, nodeID)
+		if err != nil {
+			log.WithFields(log.Fields{"nodeID": nodeID,
+				"error": err}).Error("Failed to start ETCD Cluster")
+			return err
+		}
+		return nil
+	}
+
+	// if manager, first find out who's leader, then proceed to join ETCD cluster
+	nodes, err := cli.NodeList(ctx, types.NodeListOptions{})
+	if err != nil {
+		log.WithFields(log.Fields{"nodeID": nodeID,
+			"error": err}).Error("Failed get NodeList from swarm manager")
+		return err
+	}
+	for _, n := range nodes {
+		if n.ManagerStatus != nil && n.ManagerStatus.Leader == true {
+			log.WithFields(
+				log.Fields{"leader ID": n.ID,
+					"manager ID": nodeID},
+			).Info("Found leader for swarm manager")
+
+			joinEtcdCluster(addr, n.ManagerStatus.Addr, nodeID)
+			return nil
+		}
+	}
+
+	err = fmt.Errorf("Failed to get leader for swarm manager %s", nodeID)
+	return err
+}
+
+// startEtcdCluster function is called by swarm leader to start a ETCD cluster
+func startEtcdCluster(nodeAddr string, nodeID string) error {
+	lines := []string{
+		"--name", nodeID,
+		"--advertise-client-urls", etcdScheme + nodeAddr + etcdClientPort,
+		"--initial-advertise-peer-urls", etcdScheme + nodeAddr + etcdPeerPort,
+		"--listen-client-urls", etcdScheme + etcdListenURL + etcdClientPort,
+		"--listen-peer-urls", etcdScheme + etcdListenURL + etcdPeerPort,
+		"--initial-cluster-token", etcdClusterToken,
+		"--initial-cluster", nodeID + "=" + etcdScheme + nodeAddr + etcdPeerPort,
+		"--initial-cluster-state", etcdClusterStateNew,
+	}
+	go etcdService(lines)
+
+	return nil
+}
+
+// joinEtcdCluster function is called by a non-leader swarm manager to join a ETCD cluster
+func joinEtcdCluster(nodeAddr string, leaderAddr string, nodeID string) error {
+	cli, err := addrToEtcdClient(leaderAddr)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"nodeAddr": nodeAddr,
+				"leaderAddr": leaderAddr,
+				"nodeID":     nodeID},
+		).Error("Failed to join ETCD cluster on manager ")
+	}
+
+	// list all current ETCD members, check if this node is already added as a member
+	lresp, err := cli.MemberList(context.Background())
+	if err != nil {
+		log.WithFields(
+			log.Fields{"leaderAddr": leaderAddr,
+				"error":       err,
+				"members len": len(lresp.Members)},
+		).Error("Failed to list member for ETCD")
+		return err
+	}
+
+	peerAddr := etcdScheme + nodeAddr + etcdPeerPort
+	existing := false
+	for _, member := range lresp.Members {
+		if member.PeerURLs[0] == peerAddr {
```

Comment:
Etcd clientv3 only have 4 member API functions:
https://github.com/coreos/etcd/blob/d51d381eca406156fd7e44118d4b0dab03589354/clientv3/cluster.go#L31
And there is no API function to locate a certain member...